db/table_cache.cc

```diff
@@ -175,9 +175,11 @@ InternalIterator* TableCache::NewIterator(
   PERF_TIMER_GUARD(new_table_iterator_nanos);
 
   Status s;
+  Cache::Handle* handle = nullptr;
   if (range_del_agg != nullptr && !options.ignore_range_deletions) {
     std::unique_ptr<InternalIterator> range_del_iter(NewRangeDeletionIterator(
-        options, icomparator, fd, file_read_hist, skip_filters, level));
+        options, icomparator, fd, file_read_hist, skip_filters, level,
+        &handle));
```

Comment:
Can we simply move lines from 178 to 185 to after line 226 and just call table_reader->NewRangeTombstoneIterator from there?